exp/api/v1beta1/gcpmanagedcontrolplane_types.go

```diff
@@ -27,13 +27,241 @@ const (
+type WorkloadIdentityConfig struct {
+	// Enable is to enable the workload identity config.
+	// +optional
+	Enable bool `json:"enable,omitempty"`
+
+	// WorkloadPool is the workload pool to attach all Kubernetes service accounts to Google Cloud services.
+	// Only relevant when enabled is true
+	// +optional
+	WorkloadPool string `json:"workloadPool,omitempty"`
+}
+
+// AuthenticatorGroupConfig is RBAC security group for use with Google security groups in Kubernetes RBAC.
+type AuthenticatorGroupConfig struct {
+	// Enable is to enable the authenticator group config.
+	// +optional
+	Enable bool `json:"enable,omitempty"`
+
+	// SecurityGroups is the name of the security group-of-groups to be used.
+	// +optional
+	SecurityGroups string `json:"securityGroups,omitempty"`
+}
+
+// ClusterSecurity defines the cluster security.
+type ClusterSecurity struct {
+	// WorkloadIdentityConfig allows workloads in your GKE clusters to impersonate Identity and Access Management (IAM)
+	// service accounts to access Google Cloud services
+	// +optional
+	WorkloadIdentityConfig WorkloadIdentityConfig `json:"workloadIdentityConfig,omitempty"`
+
+	// AuthenticatorGroupConfig is RBAC security group for use with Google security groups in Kubernetes RBAC.
+	// +optional
+	AuthenticatorGroupConfig AuthenticatorGroupConfig `json:"authenticatorGroupConfig,omitempty"`
+
+	// EnableLegacyAuthorization Whether the legacy (ABAC) authorizer is enabled for this cluster.
+	// +optional
+	EnableLegacyAuthorization bool `json:"enableLegacyAuthorization,omitempty"`
+
+	// IssueClientCertificate is weather to issue a client certificate.
+	// +optional
+	IssueClientCertificate bool `json:"issueClientCertificate,omitempty"`
+}
+
+// AddonsConfig defines the enabled Cluster Addons.
+type AddonsConfig struct {
+	// CloudRun enable the Cloud Run addon, which allows  the user to use a managed Knative service.
+	// +optional
+	CloudRun bool `json:"cloudRun,omitempty"`
+
+	// KalmConfig enable the KALM addon, which manages the lifecycle of k8s applications.
+	// +optional
+	KalmConfig bool `json:"kalmConfig,omitempty"`
+
+	// GKEBackup whether the Backup for GKE agent is enabled for this cluster.
+	// +optional
+	GKEBackup bool `json:"GKEBackup,omitempty"`
+
+	// GCEPersistentDiskCsiDriver whether the Compute Engine PD CSI driver is enabled for this cluster.
+	// +optional
+	GCEPersistentDiskCsiDriver bool `json:"GCEPersistentDiskCsiDriver,omitempty"`
+
+	// GCPFileStoreCsiDriver whether the GCP Filestore CSI driver is enabled for this cluster.
+	// +optional
+	GCPFileStoreCsiDriver bool `json:"GCPFileStoreCsiDriver,omitempty"`
+
+	// ImageStreaming whether to use GCFS (Google Container File System).
+	// +optional
+	ImageStreaming bool `json:"ImageStreaming,omitempty"`
+}
+
+// LoggingConfig defines the logging on Cluster.
+type LoggingConfig struct {
+	// Enable define whether enable logging to cluster or not.
+	// +kubebuilder:default=true
```

Comment:
No need to enable logging by default. Let user enable it explicitly if they wants to. Logging and monitoring operations are [priced](https://cloud.google.com/stackdriver/pricing).
 Can we remove this marker?